src/widget/radio.rs

```diff
@@ -111,6 +111,10 @@ impl<T: Data + PartialEq> Widget<T> for RadioRaw<T> {
         data: &T,
         env: &Env,
     ) -> Size {
+        if log::log_enabled!(log::Level::Warn) {
```

Comment:
I like these, but I think I would consider trying to make it more concise. Some things to consider:

- is `log_enabled` necessary? how expensive is the check?
- can we do the `log_enabled` check _inside_ the `check()` fn? Maybe we can mark that fn as `#[inline]`?
- is `Warn` the correct level? I might expect `Warn` to be turned on in release builds. If that's the case, it isn't totally clear in what situation this check will fail.

Reply:
well logs aren't ever really going to be shown a typical user, because the actual user will be using a packaged platform build, with no console; if logging is enabled it will be writing log files only for the purpose of, say, debugging a crash.

the usefulness of `log_enabled!` is that you can use it to do some expensive calculation that you don't want to do if it isn't necessary.

So what I would probably do is rename `check` to `debug_check` or something, to make it clear that it's a diagnostic, and just put it everywhere without worrying too much about the cost, assuming it isn't doing a bunch of expensive work?